packages/discount-code-generator/src/cli.js

```diff
@@ -144,13 +144,20 @@ const resolveInput = (_args) => {
         .on('data', (data) => {
           const arrayDelim = _args.multiValueDelimiter
           // Add condition so module doesn't fail if there are no cartDiscounts
-          if (data.cartDiscounts)
-          // eslint-disable-next-line no-param-reassign
+          if (data.cartDiscounts) {
+            // eslint-disable-next-line no-param-reassign
             data.cartDiscounts = data.cartDiscounts.split(arrayDelim)
-          _attributes += data
+            // eslint-disable-next-line no-param-reassign
+            data.cartDiscounts = data.cartDiscounts.map(cartDiscount => ({
+              typeId: 'cart-discount',
+              id: cartDiscount,
+            }),
+            )
+          }
+          _attributes += JSON.stringify(data)
```

Comment:
The generator expects an object, not an array. Although this implementation can also be changed. Also, I was setting it equal to the object on data, but changed it due to [this](https://github.com/commercetools/nodejs/pull/182#discussion_r119302889) review